src/en/dynasty/src/eu/kanade/tachiyomi/extension/en/dynasty/Dynasty.kt

```diff
@@ -0,0 +1,698 @@
+                    chapterName += tags.filter { it.type == "Author" }
+                        .joinToString(prefix = " by ", separator = " and ") { it.name }
+                }
+                SChapter.create().apply {
+                    url = "/chapters/$permalink"
+                    name = chapterName
+                    scanlator = tags.filter { it.type == "Scanlator" }.joinToString { it.name }
+                    date_upload = dateFormat.tryParse(releasedOn)
+                }.also(chapterList::add)
+            }
+        }
+
+        return if (data.type != "Doujin") {
+            chapterList.asReversed()
+        } else {
+            chapterList
+        }
+    }
+
+    override fun getChapterUrl(chapter: SChapter): String {
+        return baseUrl + chapter.url
+    }
+
+    override fun pageListRequest(chapter: SChapter): Request {
+        val url = chapter.url
+            .removeSuffix("/")
+
+        assert(
+            url.count { it == '/' } == 2 &&
+                url.startsWith("/chapters/"),
+        ) {
+            "Refresh Chapter List"
+        }
+
+        return GET("$baseUrl${chapter.url}.json", headers)
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        val data = response.parseAs<ChapterResponse>()
+
+        return data.pages.mapIndexed { index, page ->
+            Page(index, imageUrl = baseUrl + page.url)
+        }
+    }
+
+    override fun setupPreferenceScreen(screen: PreferenceScreen) {
+        ListPreference(screen.context).apply {
+            key = CHAPTER_FETCH_LIMIT_PREF
+            title = "Chapters Fetch Limit"
+            entries = CHAPTER_FETCH_LIMITS.map { "$it pages" }.toTypedArray()
+            entryValues = CHAPTER_FETCH_LIMITS
+            setDefaultValue(CHAPTER_FETCH_LIMITS[0])
+            summary = """
+                Limits how many pages of an entry are fetched for chapter list
+                Mostly applies to Doujins
+
+                More pages mean slower loading of chapter list
+
+                Currently fetching %s
+            """.trimIndent()
+        }.also(screen::addPreference)
+    }
+
+    private val SharedPreferences.chapterFetchLimit: Int
+        get() = getString(CHAPTER_FETCH_LIMIT_PREF, CHAPTER_FETCH_LIMITS[0])!!.let {
+            if (it == "all") {
+                Int.MAX_VALUE
+            } else {
+                it.toInt()
+            }
+        }
+
+    private val covers: Map<String, Map<String, String>> by lazy {
+        this::class.java
+            .getResourceAsStream("/assets/covers.json")!!
+            .bufferedReader().use { it.readText() }
+            .parseAs()
+    }
+
+    private fun getCoverUrl(directory: String?, permalink: String): String? {
+        directory ?: return null
+
+        if (directory == "chapters") {
+            return buildChapterCoverFetchUrl(permalink)
+        }
+
+        val file = covers[directory]?.get(permalink)
+            ?: return null
```

Comment:
What is the intended fallback? Regular updates to `convers.json`? Could you use some less efficient per-entry request instead?